cdi-microservice-provider/src/main/java/io/silverware/microservices/providers/cdi/internal/MicroserviceProxyAnnotationLiteral.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * -----------------------------------------------------------------------\
+ * SilverWare
+ *  
+ * Copyright (C) 2016 the original author or authors.
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -----------------------------------------------------------------------/
+ */
+package io.silverware.microservices.providers.cdi.internal;
+
+import io.silverware.microservices.annotations.MicroserviceProxy;
+
+import javax.enterprise.inject.spi.InjectionPoint;
+import javax.enterprise.util.AnnotationLiteral;
+
+/**
+ * Implementation of {@link io.silverware.microservices.annotations.MicroserviceProxy} for given {@link javax.enterprise.inject.spi.InjectionPoint}.
+ */
+public class MicroserviceProxyAnnotationLiteral extends AnnotationLiteral<MicroserviceProxy> implements MicroserviceProxy {
+
+   private final String bean;
+   private final String field;
+
+   public MicroserviceProxyAnnotationLiteral(final InjectionPoint injectionPoint) {
+      this.bean = injectionPoint.getBean().getName();
+      this.field = injectionPoint.getMember().getName();
```

Comment:
It shoud be. Anyway, I do not think linking it to a bean class is a good idea since there may be multiple instances of the same class and they will for sure contain the same field.